internal/resolver/dns/dns_resolver.go

```diff
@@ -183,6 +191,60 @@ type dnsResolver struct {
 	// has data race with replaceNetFunc (WRITE the lookup function pointers).
 	wg                   sync.WaitGroup
 	disableServiceConfig bool
+
+	polling chan struct{}
+}
+
+// poll begins or ends asynchronous polling of the resolver based on whether
+// err is ErrBadResolverState.
+func (d *dnsResolver) poll(errorFromClientConn error) {
+	// The only time to stop polling is when we don't receive an ErrBadResolverState from client.
+	if errorFromClientConn != balancer.ErrBadResolverState {
```

Comment:
Do you want me to filter out then? The first sentence seems to imply no? I'm going to then change to == nil.

Reply:
Do you want me to add a filter in the channel*?

Reply:
I think we should continue filtering in the channel, to avoid making unnecessary changes as part of this work.  We might want to consider removing it in the future; this magic error may have been excessive future-proofing.  But the resolver shouldn't have to care.  If `UpdateState` returns any error, assume it was *because* of the resolver's data.

Reply:
Added "filter" to channel.